core/dbt/config/selectors.py

```diff
@@ -198,8 +193,7 @@ def parse_from_selectors_list(cls, selectors):
         for selector in selectors:
             sel_name = selector["name"]
             selector_dict[sel_name] = selector
-            definition = cls.parse_from_definition(
-                selector["definition"], selector_dict=deepcopy(selector_dict)
+            selector_dict[sel_name]["definition"] = cls.parse_from_definition(
+                selector["definition"]
             )
```

Comment:
The selector_dict doesn't initially resolve the selectors.  This is left until the end to allow for it to be more dynamic